cbc-repeated-iv/src/cbc_game.rs

```diff
@@ -0,0 +1,528 @@
+        let (formatted_blocks_ct, formatted_known_ct) =
+            CBCGame::format_blocks_hex(&self.known_ciphertext);
+
+        println!("Getting corresponding known CipherText to leaked PlainText:");
+        println!("Here's leaked plaintext:");
+        println!("{formatted_blocks_pt}");
+        println!("{formatted_known_pt}\n");
+        println!("Here's corresponding CipherText to leaked PlainText:");
+        println!("{formatted_blocks_ct}");
+        println!("{formatted_known_ct}\n\n");
+
+        println!("Let's wait and eavesdrop other CipherTexts:");
+        for eavesdropped_ciphertext in &self.ciphertexts_vec {
+            let formatted_ct = CBCGame::format_blocks_hex(eavesdropped_ciphertext).1;
+            println!("{formatted_ct}");
+        }
+    }
+
+    fn example_eavesdropped_package(&self, iter: usize) {
+        // Stored strings with description for every iteration.
+        let (example_desc_start, example_desc_difference, ascii_art) =
+            CBCGame::get_desc_strings(iter);
+
+        // Let's take an example eavesdropped packet
+        // And decrypt it so we can assure, that our presumptions are correct
+        let eavesdropped_packet = &self.ciphertexts_vec[iter];
+        let decrypted_eavesdrop = &self
+            .decrypt_text(eavesdropped_packet)
+            .expect("Can not fail, as we are decrypting value, that we had encrypted");
+
+        // Do some formatting staff
+        // Format and prettify Known CT and PT
+        // Also do the same procedures to an eavesdropped packets
+        let (formatted_blocks_pt, formatted_known_pt) =
+            CBCGame::format_blocks_utf8(&self.known_plaintext);
+        let (formatted_blocks_ct, formatted_known_ct) =
+            CBCGame::format_blocks_hex(&self.known_ciphertext);
+        let formatted_eavesdropped_ct = CBCGame::format_blocks_hex(eavesdropped_packet).1;
+        let formatted_decrypted_eavesdrop = CBCGame::format_blocks_utf8(decrypted_eavesdrop).1;
+        let (blocks_status_utf8, blocks_status_hex) = self.get_blocks_status(iter);
+
+        // Let's create ascii tables for each step
+        let ascii_table_step0 = CBCGame::create_table(
+            &[
+                &"Given variables:".to_string(),
+                &format!(
+                    "{}  - known PlainText",
+                    String::from_utf8_lossy(&self.known_plaintext)
+                ),
+                &format!(
+                    "{}  - known CipherText",
+                    hex::encode(&self.known_ciphertext)
+                ),
+                &format!(
+                    "{}  - eavesdropped CipherText",
+                    hex::encode(eavesdropped_packet)
+                ),
+            ],
+            &[
+                16,
+                String::from_utf8_lossy(&self.known_plaintext).len() + 19,
+                hex::encode(&self.known_ciphertext).len() + 20,
+                hex::encode(eavesdropped_packet).len() + 27,
```

Comment:
Hard-coded constant are generally a bad sign. In this small demo they could be okay-ish. But, I would suggest you  to look and try out at some ascii table libraries. They, as I remember, support colors, utf8 chars (even emojis) and are generally easy to use.